File: scripts/payment_api.py
#!/usr/bin/env python3
"""
payment_api.py - 调用收款码 API 获取 tradeCode

用法：
    python payment_api.py --amount 50 [--order_type tip] [--payee "主播小明"] [--description "打赏主播小明50元"]
    python payment_api.py --query_reqno SEQ2033387515024719873

输出（成功）：
    SUCCESS
    reqNo=SEQ2033387515024719873
    tradeCode=6662033387515024719872
    tradeLink=https://pay-h5.4199191.xyz/payment?reqNo=SEQ2033387515024719873

输出（失败）：
    FAIL
    message=错误原因描述

输出（查询已支付）：
    QUERY_PAID
    paymentTime=2026-03-16 11:44:28

输出（查询未支付）：
    QUERY_UNPAID

输出（查询失败）：
    QUERY_FAIL
    message=错误原因描述
"""

import argparse
import json
import sys
import urllib.error
import urllib.parse
import urllib.request

API_URL = "https://pay.4199191.xyz/code"
QUERY_URL = "https://pay.4199191.xyz/payment/query"
TIMEOUT = 30


def do_query(req_no):
    url = QUERY_URL + "?" + urllib.parse.urlencode({"reqNo": req_no})
    try:
        req = urllib.request.Request(url, method="GET")
        with urllib.request.urlopen(req, timeout=TIMEOUT) as resp:
            raw = resp.read().decode("utf-8")
    except urllib.error.HTTPError as e:
        try:
            raw = e.read().decode("utf-8")
        except Exception:
            print("QUERY_FAIL")
            print("message=支付查询请求失败，请稍后重试")
            sys.exit(1)
    except urllib.error.URLError:
        print("QUERY_FAIL")
        print("message=网络连接失败，请检查网络后重试")
        sys.exit(1)
    except TimeoutError:
        print("QUERY_FAIL")
        print("message=请求超时，请稍后重试")
        sys.exit(1)
    except Exception:
        print("QUERY_FAIL")
        print("message=查询服务暂时不可用，请稍后重试")
        sys.exit(1)

    try:
        result = json.loads(raw)
    except Exception:
        print("QUERY_FAIL")
        print("message=响应解析失败，请稍后重试")
        sys.exit(1)

    result_code = result.get("resultCode")
    if result_code == 1:
        data = result.get("data", {})
        paid = data.get("paid", False)
        if paid:
            payment_time = data.get("paymentTime", "")
            print("QUERY_PAID")
            print(f"paymentTime={payment_time}")
        else:
            print("QUERY_UNPAID")
        sys.exit(0)
    else:
        message = result.get("message", "未知错误")
        print("QUERY_FAIL")
        print(f"message={message}")
        sys.exit(1)


def main():
    parser = argparse.ArgumentParser(description="调用收款码 API")
    parser.add_argument("--amount", type=float, default=None)
    parser.add_argument("--order_type", type=str, default=None)
    parser.add_argument("--payee", type=str, default=None)
    parser.add_argument("--description", type=str, default=None)
    parser.add_argument("--query_reqno", type=str, default=None)
    args = parser.parse_args()

    if args.query_reqno is not None:
        do_query(args.query_reqno)
        return

    payload = {}
    if args.amount is not None:
        payload["amount"] = args.amount
    if args.order_type is not None:
        payload["order_type"] = args.order_type
    if args.payee is not None:
        payload["payee"] = args.payee
    if args.description is not None:
        payload["description"] = args.description

    try:
        body = json.dumps(payload, ensure_ascii=False).encode("utf-8")
        req = urllib.request.Request(
            API_URL,
            data=body,
            headers={"Content-Type": "application/json"},
            method="POST",
        )
        with urllib.request.urlopen(req, timeout=TIMEOUT) as resp:
            raw = resp.read().decode("utf-8")
    except urllib.error.HTTPError as e:
        try:
            raw = e.read().decode("utf-8")
        except Exception:
            print("FAIL")
            print("message=支付服务请求失败，请稍后重试")
            sys.exit(1)
    except urllib.error.URLError:
        print("FAIL")
        print("message=网络连接失败，请检查网络后重试")
        sys.exit(1)
    except TimeoutError:
        print("FAIL")
        print("message=请求超时，请稍后重试")
        sys.exit(1)
    except Exception:
        print("FAIL")
        print("message=支付服务暂时不可用，请稍后重试")
        sys.exit(1)

    try:
        result = json.loads(raw)
    except Exception:
        print("FAIL")
        print("message=响应解析失败，请稍后重试")
        sys.exit(1)

    result_code = result.get("resultCode")
    if result_code == 1:
        data = result.get("data", {})
        req_no = data.get("reqNo", "")
        trade_code = data.get("tradeCode", "")
        trade_link = data.get("tradeLink", "")
        print("SUCCESS")
        print(f"reqNo={req_no}")
        print(f"tradeCode={trade_code}")
        print(f"tradeLink={trade_link}")
        sys.exit(0)
    else:
        message = result.get("message", "未知错误")
        print("FAIL")
        print(f"message={message}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: scripts/payment_api.js
#!/usr/bin/env node
/**
 * payment_api.js - 调用收款码 API 获取 tradeCode（Node.js 版本，仅使用内置模块）
 *
 * 用法：
 *   node payment_api.js --amount 50 [--order_type tip] [--payee "主播小明"] [--description "打赏主播小明50元"]
 *   node payment_api.js --query_reqno SEQ2033387515024719873
 *
 * 输出（成功）：
 *   SUCCESS
 *   reqNo=SEQ2033387515024719873
 *   tradeCode=6662033387515024719872
 *   tradeLink=https://pay-h5.4199191.xyz/payment?reqNo=SEQ2033387515024719873
 *
 * 输出（失败）：
 *   FAIL
 *   message=错误原因描述
 *
 * 输出（查询已支付）：
 *   QUERY_PAID
 *   paymentTime=2026-03-16 11:44:28
 *
 * 输出（查询未支付）：
 *   QUERY_UNPAID
 *
 * 输出（查询失败）：
 *   QUERY_FAIL
 *   message=错误原因描述
 */

const https = require('https');
const http = require('http');

const API_URL = 'https://pay.4199191.xyz/code';
const QUERY_URL = 'https://pay.4199191.xyz/payment/query';
const TIMEOUT_MS = 30000;

function parseArgs() {
  const args = process.argv.slice(2);
  const result = {};
  for (let i = 0; i < args.length; i++) {
    if (args[i].startsWith('--') && i + 1 < args.length) {
      const key = args[i].slice(2);
      result[key] = args[i + 1];
      i++;
    }
  }
  return result;
}

function fail(message) {
  process.stdout.write('FAIL\n');
  process.stdout.write(`message=${message}\n`);
  process.exit(1);
}

function queryFail(message) {
  process.stdout.write('QUERY_FAIL\n');
  process.stdout.write(`message=${message}\n`);
  process.exit(1);
}

function success(reqNo, tradeCode, tradeLink) {
  process.stdout.write('SUCCESS\n');
  process.stdout.write(`reqNo=${reqNo}\n`);
  process.stdout.write(`tradeCode=${tradeCode}\n`);
  process.stdout.write(`tradeLink=${tradeLink}\n`);
  process.exit(0);
}

function doQuery(reqNo) {
  const url = new URL(QUERY_URL + '?reqNo=' + encodeURIComponent(reqNo));
  const lib = url.protocol === 'https:' ? https : http;

  const options = {
    hostname: url.hostname,
    port: url.port || (url.protocol === 'https:' ? 443 : 80),
    path: url.pathname + url.search,
    method: 'GET',
    headers: {
      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/145.0.0.0 Safari/537.36',
    },
  };

  const req = lib.request(options, (res) => {
    let raw = '';
    res.setEncoding('utf8');
    res.on('data', (chunk) => { raw += chunk; });
    res.on('end', () => {
      let result;
      try {
        result = JSON.parse(raw);
      } catch (e) {
        queryFail('响应解析失败，请稍后重试');
        return;
      }
      if (result.resultCode === 1) {
        const data = result.data || {};
        if (data.paid) {
          process.stdout.write('QUERY_PAID\n');
          process.stdout.write(`paymentTime=${data.paymentTime || ''}\n`);
          process.exit(0);
        } else {
          process.stdout.write('QUERY_UNPAID\n');
          process.exit(0);
        }
      } else {
        queryFail(result.message || '未知错误');
      }
    });
  });

  req.setTimeout(TIMEOUT_MS, () => {
    req.destroy();
    queryFail('请求超时，请稍后重试');
  });

  req.on('error', (e) => {
    if (e.code === 'ECONNREFUSED' || e.code === 'ENOTFOUND') {
      queryFail('网络连接失败，请检查网络后重试');
    } else {
      queryFail('查询服务暂时不可用，请稍后重试');
    }
  });

  req.end();
}

const args = parseArgs();

if (args.query_reqno !== undefined) {
  doQuery(args.query_reqno);
} else {
  const payload = {};
  if (args.amount !== undefined) payload.amount = parseFloat(args.amount);
  if (args.order_type !== undefined) payload.order_type = args.order_type;
  if (args.payee !== undefined) payload.payee = args.payee;
  if (args.description !== undefined) payload.description = args.description;

  const body = JSON.stringify(payload);
  const url = new URL(API_URL);
  const lib = url.protocol === 'https:' ? https : http;

  const options = {
    hostname: url.hostname,
    port: url.port || (url.protocol === 'https:' ? 443 : 80),
    path: url.pathname,
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Content-Length': Buffer.byteLength(body),
      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/145.0.0.0 Safari/537.36',
    },
  };

  const req = lib.request(options, (res) => {
    let raw = '';
    res.setEncoding('utf8');
    res.on('data', (chunk) => { raw += chunk; });
    res.on('end', () => {
      let result;
      try {
        result = JSON.parse(raw);
      } catch (e) {
        fail('响应解析失败，请稍后重试');
        return;
      }
      if (result.resultCode === 1) {
        const data = result.data || {};
        success(data.reqNo || '', data.tradeCode || '', data.tradeLink || '');
      } else {
        fail(result.message || '未知错误');
      }
    });
  });

  req.setTimeout(TIMEOUT_MS, () => {
    req.destroy();
    fail('请求超时，请稍后重试');
  });

  req.on('error', (e) => {
    if (e.code === 'ECONNREFUSED' || e.code === 'ENOTFOUND') {
      fail('网络连接失败，请检查网络后重试');
    } else {
      fail('支付服务暂时不可用，请稍后重试');
    }
  });

  req.write(body);
  req.end();
}
